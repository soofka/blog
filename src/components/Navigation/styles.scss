@import 'components/Blog/variables.scss';

.navigation {
  margin: $marginM $marginS;
  padding: 0;

  li {
    display: block;
    margin: 0;
    padding: $paddingS $paddingXS;
    font-size: $fontSizeM;
    line-height: $fontSizeL;
    font-weight: bold;
  }
}

@media (max-width: $mobileBreakPointWidth) {
  .navigation {
    float: left;

    li {
      display: inline-block;
    }
  }
}
