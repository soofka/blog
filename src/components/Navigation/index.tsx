import * as React from 'react';
import { Link } from 'react-router-dom';

import './styles.scss';

import RoutingProvider from 'common/RoutingProvider';

export const Navigation = (): JSX.Element => {
  return (
    <div>
      <ul className="navigation">
        <Link to={RoutingProvider.getHomeNiceUrlBasePath()}><li>Home</li></Link>
        <Link to={RoutingProvider.getAboutNiceUrlBasePath()}><li>About</li></Link>
      </ul>
    </div>
  );
};

export default Navigation;
