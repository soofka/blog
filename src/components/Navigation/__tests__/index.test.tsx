import * as React from 'react';
import { shallow } from 'enzyme';

import Navigation from 'components/Navigation';

describe('<Navigation/>', () => {

  it('renders correctly', () => {
    expect(shallow(<Navigation/>)).toMatchSnapshot();
  });

});
