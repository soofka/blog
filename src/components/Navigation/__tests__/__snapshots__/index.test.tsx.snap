exports[`<Navigation/> renders correctly 1`] = `
<div>
  <ul
    className="navigation">
    <Link
      replace={false}
      to="/">
      <li>
        Home
      </li>
    </Link>
    <Link
      replace={false}
      to="/about">
      <li>
        About
      </li>
    </Link>
  </ul>
</div>
`;
